### DISCUSSION TITLE
[v8] different global filter fn per column

### DISCUSSION THREAD

**Discussion by chiptus** [TanStack/table#4505] [Q&A]
if some of my columns have different values than string, I need to use a custom global filter. see https://codesandbox.io/s/beautiful-roentgen-qoqp65?file=/src/main.tsx

would be helpful to be able to set different global functions per column.

or is there another way?

**Comment by ArtemFedorchuk** [comment]
> if some of my columns have different values than string, I need to use a custom global filter. see https://codesandbox.io/s/beautiful-roentgen-qoqp65?file=/src/main.tsx
> 
> would be helpful to be able to set different global functions per column.
> 
> or is there another way?

Have you found some way how to use globalFilter or not?
Because I need to have a search field for searching by columns.

**Comment by pi-kei** [comment]
You can write you custom global filter like that: 

```ts
const myGlobalFilter: FilterFn<any> = function (
  row,
  columnId,
  filterValue,
  addMeta
) {
  const value = row.getValue(columnId);
  // filter per type
  if (typeof value === 'string') return filterFns.includesString(row, columnId, filterValue, addMeta);
  if (typeof value === 'number') return filterFns.weakEquals(row, columnId, filterValue, addMeta);
  ...
}
```
You can do filter per columnId the same way.
